# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSinkWrapper.cpp
# Commit: 41d154330f7f
# Full Hash: 41d154330f7f72ef2bd6b66e592e68064d97950f
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-10 03:50:31
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Trigger the audibility event at the beginning of the initialization of an audio stream, when un-muting. r=alwu
#   
#   The initialization itself can take some time and this audibility event is used
#   to prioritize the process, that will soon become audible, on at least Windows,
#   so it's better to get prioritized slightly ahead of time.
# ==============================================================================

diff -r f09bb9e00c17 -r 41d154330f7f dom/media/mediasink/AudioSinkWrapper.cpp
--- a/dom/media/mediasink/AudioSinkWrapper.cpp	Mon May 09 13:20:54 2022 +0000
+++ b/dom/media/mediasink/AudioSinkWrapper.cpp	Mon May 09 13:20:54 2022 +0000
@@ -325,7 +325,8 @@
           // for that amount of time, and the video would therefore not
           // update. The Start() call is very cheap on the other hand, we can
           // do it from the MDSM thread.
-          nsresult rv = audioSink->InitializeAudioStream(mParams);
+          nsresult rv = audioSink->InitializeAudioStream(mParams,
+              AudioSink::InitializationType::UNMUTING);
           mOwnerThread->Dispatch(NS_NewRunnableFunction(
               "StartAudioSink (Async part: start from MDSM thread)",
               [self = RefPtr<AudioSinkWrapper>(this),
@@ -382,7 +383,8 @@
         }));
   } else {
     mAudioSink.reset(mCreator->Create());
-    nsresult rv = mAudioSink->InitializeAudioStream(mParams);
+    nsresult rv = mAudioSink->InitializeAudioStream(mParams,
+        AudioSink::InitializationType::INITIAL);
     if (NS_FAILED(rv)) {
       mEndedPromiseHolder.RejectIfExists(rv, __func__);
       LOG("Sync AudioSinkWrapper initialization failed");
